### PR TITLE
prisma,prisma-engines: 5.22.0 -> 6.0.1

### PR DESCRIPTION
**Pull request by aqrln**
- **prisma-engines: 5.22.0 -> 6.0.1**
- **prisma: 5.22.0 -> 6.0.1**

* https://github.com/prisma/prisma/releases/tag/6.0.0
* https://github.com/prisma/prisma/releases/tag/6.0.1

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by aqrln** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>documenso</li>
    <li>pingvin-share.backend</li>
    <li>prisma</li>
    <li>prisma-engines</li>
  </ul>
</details>

**Review by tomhoule — APPROVED** [review]
noice!

File: pkgs/by-name/pr/prisma/package.nix [hunk 1]
```diff
@@ -32,7 +32,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   pnpmDeps = pnpm_8.fetchDeps {
     inherit (finalAttrs) pname version src;
-    hash = "sha256-2o6ClY0zMctLR4nFmApiYnzXlrN1EqbHkAP/FEcXnEQ=";
+    hash = "sha256-fOg32w/fQkyn8HBMffUKob7XzOQLtsB642pDdEz/y2E=";
   };
```
Comment (Scrumplex) [inline pkgs/by-name/pr/prisma/package.nix:36]: Could you replace pnpm_8 with pnpm_9 and regenerate hashes?

File: pkgs/development/tools/database/prisma-engines/default.nix [hunk 2]
```diff
@@ -28,7 +28,7 @@ rustPlatform.buildRustPackage rec {
     lockFile = ./Cargo.lock;
```
Comment (Scrumplex) [inline pkgs/development/tools/database/prisma-engines/default.nix:28]: Could you change this to use `fetchCargoVendor` that was introduced in https://github.com/NixOS/nixpkgs/pull/349360

File: nixos/doc/manual/release-notes/rl-2505.section.md [hunk 3]
```diff
@@ -52,6 +52,11 @@
   add `vimPlugins.notmuch-vim` to your (Neo)vim configuration if you want the
   vim plugin.
 
+- `prisma` and `prisma-engines` have been updated to version 6.0.0, which
+  introduces several breaking changes. See the [Prisma ORM upgrade
+  guide](https://www.prisma.io/docs/orm/more/upgrade-guides/upgrading-versions/upgrading-to-prisma-6)
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2505.section.md:57]:
```suggestion
  introduces several breaking changes. See the
  [Prisma ORM upgrade guide](https://www.prisma.io/docs/orm/more/upgrade-guides/upgrading-versions/upgrading-to-prisma-6)
```
splitting links on new lines makes them hard to find again

**Comment by aqrln** [comment]
@Scrumplex @SuperSandro2000 I've addressed the feedback

**Review by tomhoule — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**